### DISCUSSION TITLE
Vlan (vcreate) not work on Dell MX750c with latest build

### DISCUSSION THREAD

**Discussion by EmilianoBregola** [ipxe/ipxe#842] [Q&A]
Hi, I am new of this project and had start to know ipxe only from last 3 days but have a problem with a potentially lager enviroment of server where would like to configure unattended install of linux enviroment.

the network configuration of us enviroment need that in all the server we need to set the vlan that come everywhere in trunk mode
we want to initialize pxe boot in uefi configuration and need to EMBED a script to setup vlan tag in ipxe.efi boot loader.
we have already configure dhcpd and it work fine.
making new ipxe.efi bootloader have problem. the PXE boot initialized by bios work and get the first ip from dhcp to Biox PXE Install, after that, manually or inside compliled file the vlan creation work inside the ipxe.efi kernel, the dhcp offer a new ip to ipxe kernel but never we can ping this ip and can't get nothing from us tftp server, like tftp://x.x.x.x/menu.ipxe or other like file that are axposed by tftp server.

we do not have problem with older ipxe version that come with redhat 8.x work (the ipxe version in that case is 1.0.0+) but in this situation we do not have che source code to compile the file ipxe.efi with EMBED option to create vlan and initialize it.

Can you help me to understand where can be the problem? meaby is a driver problem?

Best Regards
Emiliano

**Comment by NiKiZe** [comment]
Searching for existing posts and you can find #369 

Why would you be using an old version of iPXE? Please try current git master.

**Reply by EmilianoBregola** [reply]
Hi NiKiZe,
thank you for your suggestion, I has in first use a precompiled ipxe that was delivery in redhat media. Now that we have understand the necessity of a EMBED parameter for vlan creation and chain use, have switch to compiling the master version, with vlan support also, but this version appear not inizialize correctly the network layer, instread the older that work,
I will try to increase debug level of vlan component,
will update you after.
best regards

**Comment by EmilianoBregola** [comment]
Hi, we has resolve us issue by applying this patch code https://github.com/ipxe/ipxe/commit/133f4c47baef6002b2ccb4904a035cda2303c6e5 
compiled in a rocky 8.x development enviroment. the complete task was:


make bin-x86_64-efi/ipxe.efi

[root@rocky8-devel pass]# git clone https://github.com/ipxe/ipxe.git
Cloning into 'ipxe'...
remote: Enumerating objects: 57478, done.
remote: Counting objects: 100% (537/537), done.
remote: Compressing objects: 100% (235/235), done.
remote: Total 57478 (delta 377), reused 392 (delta 302), pack-reused 56941
Receiving objects: 100% (57478/57478), 18.29 MiB | 4.35 MiB/s, done.
Resolving deltas: 100% (42921/42921), done.

[root@rocky8-devel pass]# cd ipxe/
[root@rocky8-devel ipxe]# cd src/
[root@rocky8-devel src]# git checkout 133f4c47
Note: switching to '133f4c47'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

 

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

 git switch -c <new-branch-name>

Or undo this operation with:

 

  git switch -

 

Turn off this advice by setting config variable advice.detachedHead to false

 

HEAD is now at 133f4c47 [build] Handle R_X86_64_PLT32 from binutils 2.31

 

[root@rocky8-devel src]# touch custom.ipxe

 



# we has enable that features (with PXE_CMD fatures enabled the make fail):
[root@rocky8-devel src]# vim config/general.h

#define NSLOOKUP_CMD          /* DNS resolving command */
#define TIME_CMD              /* Time commands */
//#define DIGEST_CMD            /* Image crypto digest commands */
#define LOTEST_CMD            /* Loopback testing commands */
#define VLAN_CMD              /* VLAN commands */
//#define PXE_CMD               /* PXE commands */
#define REBOOT_CMD            /* Reboot command */
#define POWEROFF_CMD          /* Power off command */
#define IMAGE_TRUST_CMD       /* Image trust management commands */
#define PCI_CMD               /* PCI commands */
#define PARAM_CMD             /* Form parameter commands */
#define NEIGHBOUR_CMD         /* Neighbour management commands */
#define PING_CMD              /* Ping command */
#define CONSOLE_CMD           /* Console command */
#define IPSTAT_CMD            /* IP statistics commands */
#define PROFSTAT_CMD          /* Profiling commands */
#define NTP_CMD               /* NTP commands */
//#define CERT_CMD              /* Certificate management commands */


[root@rocky8-devel src]# cat > custom.ipxe <<EOF
#!ipxe
prompt --key 0x02 --timeout 2000 Press Ctrl-B for the iPXE command line... && shell ||
vcreate -t 3037 net0
autoboot net0-3037
EOF

  

[root@rocky8-devel src]# make bin-x86_64-efi/ipxe.efi EMBED=/storage/pass/ipxe/src/custom.ipxe

 

# Generazione mini kernel per boot di iPXE
[xfly51@rocky8-devel src]$ ls ./bin-x86_64-efi/ipxe.efi
./bin-x86_64-efi/ipxe.efi



thank you for the support.
Best Regards
Emiliano

**Reply by stappersg** [reply]
Acknowlege on the _"four year old version works for me"_.

Regarding the features, that seems
```diff
--- a/src/config/general.h
+++ b/src/config/general.h
@@ -136,23 +136,23 @@ FILE_LICENCE ( GPL2_OR_LATER_OR_UBDL );
 #define LOGIN_CMD              /* Login command */
 #define SYNC_CMD               /* Sync command */
 #define SHELL_CMD              /* Shell command */
-//#define NSLOOKUP_CMD         /* DNS resolving command */
-//#define TIME_CMD             /* Time commands */
+#define NSLOOKUP_CMD           /* DNS resolving command */
+#define TIME_CMD               /* Time commands */
 //#define DIGEST_CMD           /* Image crypto digest commands */
-//#define LOTEST_CMD           /* Loopback testing commands */
-//#define VLAN_CMD             /* VLAN commands */
+#define LOTEST_CMD             /* Loopback testing commands */
+#define VLAN_CMD               /* VLAN commands */
 //#define PXE_CMD              /* PXE commands */
-//#define REBOOT_CMD           /* Reboot command */
-//#define POWEROFF_CMD         /* Power off command */
-//#define IMAGE_TRUST_CMD      /* Image trust management commands */
-//#define PCI_CMD              /* PCI commands */
-//#define PARAM_CMD            /* Form parameter commands */
-//#define NEIGHBOUR_CMD                /* Neighbour management commands */
-//#define PING_CMD             /* Ping command */
-//#define CONSOLE_CMD          /* Console command */
-//#define IPSTAT_CMD           /* IP statistics commands */
-//#define PROFSTAT_CMD         /* Profiling commands */
-//#define NTP_CMD              /* NTP commands */
+#define REBOOT_CMD             /* Reboot command */
+#define POWEROFF_CMD           /* Power off command */
+#define IMAGE_TRUST_CMD                /* Image trust management commands */
+#define PCI_CMD                        /* PCI commands */
+#define PARAM_CMD              /* Form parameter commands */
+#define NEIGHBOUR_CMD          /* Neighbour management commands */
+#define PING_CMD               /* Ping command */
+#define CONSOLE_CMD            /* Console command */
+#define IPSTAT_CMD             /* IP statistics commands */
+#define PROFSTAT_CMD           /* Profiling commands */
+#define NTP_CMD                        /* NTP commands */
 //#define CERT_CMD             /* Certificate management commands */
 
 /*
```
(Only `//` at the begin of the removed)

When time permits, please check the latest version of _iPXE_, a version that has merge request #838 included.  (Got merged five days ago.)

**Reply by NiKiZe** [reply]
Why are you doing this change to begin with?

First use git master.
And don't change configs unless you _have to_ that is after compiling and testing and having features actually missing.

Of course you can't enable PCBIOS only features such as PXE in EFI builds.

**Reply by stappersg** [reply]
And another improvement in the VLAN area was merged.
```text
Date: Thu, 22 Dec 2022 11:07:28 -0800
From: Michael Brown <notifications@github.com>
To: ipxe/ipxe <ipxe@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Subject: Re: [ipxe/ipxe] VLAN support for cached DHCPACK (PR #844)

Merged #844 into master.
--
Reply to this email directly or view it on GitHub:                                                
https://github.com/ipxe/ipxe/pull/844#event-8098441912                                                            
You are receiving this because you are subscribed to this thread.
```
